### PR TITLE
SendResultUseCase implementation

### PR DESCRIPTION
**Pull request by basiliskus**
# SendResultUseCase implementation

Initial implementation for `SendResultUseCase` + needed result interfaces

## Issue

#615 

## Checklist

- [x] I have added tests to cover my changes
- [x] I have added JavaDocs where required

### CODE REVIEW

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/messages/SendMessageUseCase.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package gov.hhs.cdc.trustedintermediary.etor.messages;
+
+/**
+ * This interface represents a use case for sending a generic message. It provides a method to
+ * convert and send the message.
+ *
+ * @param <T> the type of message to be sent
```
Comment (jcrichlake) [inline etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/messages/SendMessageUseCase.java:7]: Is this a param?

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/results/SendResultUseCase.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    @Inject MessageSender<Result<?>> sender;
+
+    private SendResultUseCase() {}
+
+    public static SendResultUseCase getInstance() {
+        return INSTANCE;
+    }
+
+    @Override
+    public void convertAndSend(Result<?> result, String receivedSubmissionId)
```
Comment (jcrichlake) [inline etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/results/SendResultUseCase.java:20]: Should the conversion happen in the use case or should that be a separate class to handle that?

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/messages/SendMessageUseCase.java (hunk 1 above)
Comment (basiliskus) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/messages/SendMessageUseCase.java:7]: Generics use the same @param tag in Javadocs: https://stackoverflow.com/a/2015998

**Review by saquino0827 — COMMENTED** [review]
I like the Message typing. I was thinking of combining the Result and Order objects to a Message object as well but I wanted to discuss with you and @jorg3lopez . I'll wait until he takes a look at this PR and we can see where we want to go but the PR looks good!

**Comment by jcrichlake** [comment]
Really nice PR! Just a couple of questions that the answer may just be "No it's fine" hah

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/results/SendResultUseCase.java (hunk 2 above)
Comment (basiliskus) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/results/SendResultUseCase.java:20]: I think it should. That's what we're doing in `SendOrderUseCase`. Maybe the use case should be named `ConvertAndSendResultUseCase` but I don't know if that's necessary

**Review by saquino0827 — APPROVED** [review]

**Pull request merged**